### PR TITLE
add a circleci integration: check unclosed tags (#4417)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #4417 to release-4.0

---

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/3786/
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,6 +20,6 @@
 
 ## 贡献文档
 
-[<img src="media/contribution-map.png" alt="contribution-map" width="180">](https://github.com/pingcap/community/tree/master/special-interest-groups/sig-docs)
+[<img src="media/contribution-map.png" alt="contribution-map" width="180"></img>](https://github.com/pingcap/community/tree/master/special-interest-groups/sig-docs)
```
Comment (yikeke) [inline README.md:23]:
```suggestion
[<img src="media/contribution-map.png" alt="contribution-map" width="180"></img](https://github.com/pingcap/community/tree/master/special-interest-groups/sig-docs)
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -30,7 +30,7 @@ jobs:
           command: |
             git remote add upstream https://github.com/pingcap/docs-cn.git
             git fetch upstream
-            markdownlint $(git diff-tree --name-only --no-commit-id -r upstream/master..HEAD -- '*.md' ':(exclude).github/*' ':(exclude)v1.0/*' ':(exclude)v2.0/*' ':(exclude)v2.1-legacy/*')
+            markdownlint $(git diff-tree --name-only --no-commit-id -r upstream/release-4.0..HEAD -- '*.md' ':(exclude).github/*' ':(exclude)v1.0/*' ':(exclude)v2.0/*' ':(exclude)v2.1-legacy/*')
```
Comment (yikeke) [inline .circleci/config.yml:33]:
```suggestion
            markdownlint $(git diff-tree --name-only --no-commit-id -r upstream/release-4.0..HEAD -- '*.md' ':(exclude).github/*')
```
 @aylei ':(exclude)v1.0/*' ':(exclude)v2.0/*' ':(exclude)v2.1-legacy/*') 这仨文件夹在本仓库已经不存在了，可以删去了不？另外由于这是 release-4.0 分支，所以我把 upstream/master..HEAD 改成了 upstream/release-4.0..HEAD。

File: .circleci/config.yml [hunk 3]
```diff
@@ -42,6 +42,11 @@ jobs:
          command: |
            scripts/verify-link-anchors.sh
 
+      - run:
+         name: "Check unclosed tags"
+         command:
+          python3 scripts/check-tags.py $(git diff-tree --name-only --no-commit-id -r upstream/release-4.0..HEAD -- '*.md' ':(exclude).github/*' ':(exclude)v1.0/*' ':(exclude)v2.0/*' ':(exclude)v2.1-legacy/*')
```
Comment (yikeke) [inline .circleci/config.yml:48]:
```suggestion
          python3 scripts/check-tags.py $(git diff-tree --name-only --no-commit-id -r upstream/release-4.0..HEAD -- '*.md' ':(exclude).github/*')
```
git diff-tree --name-only --no-commit-id -r upstream/release-4.0..HEAD -- '*.md' 的准确意思是 **当前 branch** 与 upstream/release-4.0 branch 的 md 文件的不同吗？ @aylei

File: .circleci/config.yml (hunk 2 above)
Comment (aylei) [reply .circleci/config.yml:33]: 嗯，LGTM

File: .circleci/config.yml (hunk 3 above)
Comment (aylei) [reply .circleci/config.yml:48]: 是的

Comment (yikeke) [reply .circleci/config.yml:48]: 好的，感谢

File: .circleci/config.yml (hunk 2 above)
Comment (yikeke) [reply .circleci/config.yml:33]: Thanks!

**Review by yikeke — APPROVED** [review]
LGTM

File: README.md [hunk 4]
```diff
@@ -20,6 +20,6 @@
 
 ## 贡献文档
 
-[<img src="media/contribution-map.png" alt="contribution-map" width="180">](https://github.com/pingcap/community/tree/master/special-interest-groups/sig-docs)
+[<img src="media/contribution-map.png" alt="contribution-map" width="180"></img](https://github.com/pingcap/community/tree/master/special-interest-groups/sig-docs)
```
Comment (yikeke) [inline README.md:23]:
```suggestion
[<img src="media/contribution-map.png" alt="contribution-map" width="180"></img>](https://github.com/pingcap/community/tree/master/special-interest-groups/sig-docs)
```

**Review by yikeke — APPROVED** [review]

**Pull request merged**